### ISSUE TITLE
rnaspades: finished abnormally, err code: -6 Assertion `path.Length() > 0' failed.

### ISSUE THREAD

**Issue by MDSharma** [ablab/spades#164]
Hi Folks,

Please see log extract below and attached:
````
33:00:03.315   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 11737368 (0%)
 33:00:04.135   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 11737368 (0%)
 33:00:05.473   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 11737368 (0%)
 33:00:07.257   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 11737368 (0%)
 33:00:09.864   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 11737368 (0%)
 33:00:12.932   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 11737368 (0%)
 33:00:19.968   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 8192 paths from 11737368 (0%)
 33:00:30.074   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 16384 paths from 11737368 (0%)
 33:00:48.043   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 32768 paths from 11737368 (0%)
 33:01:16.868   227G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 65536 paths from 11737368 (0%)
 33:01:58.600   228G / 447G  INFO    General                 (path_extender.hpp         : 883)   Processed 131072 paths from 11737368 (1%)
=== Stack Trace ===
[0x40ac28]
[0x67c1a5]
[0x68483e]
[0x682295]
[0x68cf2a]
[0x694aa8]
[0x694fbe]
[0x678ef7]
[0x698b67]
[0x657fd2]
[0x66c97e]
[0x6563a9]
[0x4c2f04]
[0x55d2c1]
[0x408c19]
[0x40123c]
[0x87a73b]
[0x407c01]
spades-core: /spades/src/common/modules/path_extend/weight_counter.hpp:295: virtual double path_extend::CoverageAwareIdealInfoProvider::EstimatePathCoverage(const path_extend::BidirectionalPath&) const: Assertion `path.Length() > 0' failed.


== Error ==  system call for: "['/cm/shared/apps/SPAdes/3.12.0/bin/spades-core', '/scratch/dg288-2/rnaspades/K61/configs/config.info', '/scratch/dg288-2/rnaspades/K61/configs/mda_mode.info', '/scratch/dg288-2/rnaspades/K61/configs/rna_mode.info']" finished abnormally, err code: -6
```` 

The system this was running on has 2TB of RAM and 128 cores with a native ext4 filesystem (so some of the other threads re err code -6 may not be applicable here.

Here is the params.txt
````
ommand line: /cm/shared/apps/SPAdes/3.12.0/bin/rnaspades.py    --dataset       /scratch/user/test.yaml      -m      1024    -t      100     --ss-rf -o      /scratch/user/rnaspades

System information:
  SPAdes version: 3.12.0
  Python version: 2.7.12
  OS: Linux-3.10.0-693.21.1.el7.x86_64-x86_64-with-centos-7.4.1708-Core

Output dir: /scratch/user/rnaspades
Mode: read error correction and assembling
Debug mode is turned OFF

Dataset parameters:
  RNA-seq mode
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/scratch/user/trimmed/0009_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0009_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0013_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0013_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0050_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0050_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0063_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0063_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0081_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0081_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0128_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0128_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/0143_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/0143_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4265_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4265_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4282_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4282_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4309_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4309_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4315_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4315_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4321_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4321_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4322_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4322_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4336_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4336_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4341_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4341_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4352_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4352_D14_R2_val_2.fq.gz', '/scratch/user/trimmed/4442_D03_R2_val_2.fq.gz', '/scratch/user/trimmed/4442_D14_R2_val_2.fq.gz']
      right reads: ['/scratch/user/trimmed/0009_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0009_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0013_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0013_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0050_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0050_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0063_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0063_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0081_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0081_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0128_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0128_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/0143_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/0143_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4265_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4265_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4282_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4282_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4309_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4309_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4315_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4315_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4321_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4321_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4322_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4322_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4336_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4336_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4341_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4341_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4352_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4352_D14_R1_val_1.fq.gz', '/scratch/user/trimmed/4442_D03_R1_val_1.fq.gz', '/scratch/user/trimmed/4442_D14_R1_val_1.fq.gz']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
Read error correction parameters:
  Iterations: 1
  PHRED offset will be auto-detected
  Corrected reads will be compressed
Assembly parameters:
  k: auto
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /scratch/user/rnaspades/tmp
  Threads: 100
  Memory limit (in Gb): 1024
````

.. and then the logs..

**Comment by desmondramirez** [comment]
I've gotten this same error for some of my spades runs (but not all) as well. Any ideas? To work around, I've just been trying to re-run the assemblies.

```
System information:
  SPAdes version: 3.12.0
  Python version: 3.6.5
  OS: Linux-2.6.32-696.28.1.el6.x86_64-x86_64-with-redhat-6.9-Santiago

Output dir: /project/spades_k55
Mode: ONLY assembling (without read error correction)
Debug mode is turned OFF

Dataset parameters:
  RNA-seq mode
  Reads:
    Library number: 1, library type: paired-end
      orientation: fr
      left reads: ['/project/1P.cor.fq']
      right reads: ['/project/2P.cor.fq']
      interlaced reads: not specified
      single reads: not specified
      merged reads: not specified
Assembly parameters:
  k: [55]
  Repeat resolution is enabled
  Mismatch careful mode is turned OFF
  MismatchCorrector will be SKIPPED
  Coverage cutoff is turned OFF
Other parameters:
  Dir for temp files: /project
  Threads: 8
  Memory limit (in Gb): 100

spades-core: /spades/src/common/modules/path_extend/weight_counter.hpp:295: virtual double path_extend::CoverageAwareIdealInfoProvider::EstimatePathCoverage(const path_extend::BidirectionalPath&) const: Assertion `path.Length() > 0' failed.


== Error ==  system call for: "['/project/spades-3.12.0-1/bin/spades-core', '/project/spades_k55/K55/configs/config.info', '/project/spades_k55/K55/configs/mda_mode.info', '/project/spades_k55/K55/configs/rna_mode.info']" finished abnormally, err code: -6
```

**Comment by andrewprzh** [comment]
Hello!

This error is quite unexpected for us, but unfortunately we've seen it before. Is it possible to share any of the data that results in an error so we could reproduce it?

Best
Andrey

**Comment by MDSharma** [comment]
Hi Andrey,

If you’ve seen it before, do you have any thoughts on what triggers it?

Our dataset is quite large and not published yet so can’t share without discussions with colleagues.

Cheers,
MD

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by andrewprzh** [comment]
Unfortunately, I've only seen a report from another user, who's data was also private. 

The only option I have now is to dig into the code and try to find out how this situation may happen. This is clearly a bug.

**Comment by andrewprzh** [comment]
I will post any news regarding this issue as soon as we have any.

**Comment by MDSharma** [comment]
Happy to run a debug script (e.g if you have a modified script where each step is logged or debugged to see where this crops up) on our data..

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by andrewprzh** [comment]
That would be great! I'll send you a modified version with extended log in a while.

**Comment by desmondramirez** [comment]
Hi Andrey

I can share the data that triggered this error with you. What's the best way to get it to you? Each set of reads is ~ 4.2Gb. Also, it might not matter, but I've been using Spades as part of the Oyster River Protocol (https://oyster-river-protocol.readthedocs.io/en/latest/). I don't expect that to be part of the problem though.

Let me know!
Des

**Comment by andrewprzh** [comment]
Hi Des

That would be awesome. Could you write me to andrewprzh@gmail.com and I will provide you the instructions?

Yes, you are right regarding the Oyster River Protocol. If you provide the exact command we should be able to the error.

Best
Andrey

**Comment by MDSharma** [comment]
Hello, please do let me know when you have this ready - we are eager to move this along.


In the mean time, I might re-run with normalized data (we were trying to use non-digitally-normalised data as per some publications so far)

________________________________

**Comment by andrewprzh** [comment]
Sorry for a delay -- @desmondramirez provided a dataset with the exactly same error, so I hope there will a fix soon.

Best
Andrey

**Comment by andrewprzh** [comment]
We have quick fix, which will be included in the next release. If you need the fixed version ASAP, please write me directly to andrewprzh@gmail.com and I will send you the fix.

Best
Andrey

**Comment by andrewprzh** [comment]
Fixed in SPAdes 3.13.0
